### ISSUE TITLE
install got an error, rolling back install

### ISSUE THREAD

**Issue by jplatte** [nodejs/node-gyp#690]
So, I'm a user of the Atom editor on Windows, and trying to install any package (even packages that don't require native stuff and should not depend on gyp) get me an error message. On the Atom discussion site [I was told][atom-discuss] that I might find help here. No matter what I try to install (`apm install <any package name>`), I get the following output:

    Installing <package name> to C:\Users\jplatte\.atom\packages
    gyp info it worked if it ends with ok
    gyp info using node-gyp@1.0.2
    gyp info using node@0.10.35 | win32 | ia32
    gyp http GET https://atom.io/download/atom-shell/v0.22.0/node-v0.22.0.tar.gz
    gyp WARN install got an error, rolling back install
    gyp

I don't have any experience with npm or gyp, and it is very frustrating to still get the same error message after checking for the tenth time that I have everything installed that I should need. How do I debug this problem?

[atom-discuss]: https://discuss.atom.io/t/cant-install-any-apm-packages-on-windows-7/18984/9

**Comment by Fishrock123** [comment]
> Python 2.7 installed (C:\Python27)

I'm not sure gyp likes 2.7.

Could you try installing `node-gyp` globally if it isn't already:

`npm install -g node-gyp`

Then go into that package's directory and run `node-gyp rebuild --verbose`?

**Comment by jplatte** [comment]
Finally, I found the problem. Apparently the proxy settings are lost when https://atom.io/download/atom-shell/v0.22.0/node-v0.22.0.tar.gz redirects to another page. I have tried manually building packages, but the only package I'm aware of is that file, and when I ran gyp in there it complained about a missing bindings.gyp file. I still don't even know how gyp is used in all this, but to be honest, I don't care.

Anyway, this redirection + proxy problem wasn't found by me. There's a bug report about the issue I had on the apm repo (that I didn't find before because the error message I got was too vague). I don't know if anyone has actually filed a bug report here instead of just posting workarounds, so here's a link: https://github.com/atom/apm/issues/322
